lib/optimize/ConcatenatedModule.js

```diff
@@ -297,6 +294,18 @@ class ConcatenatedModule extends Module {
 		return list;
 	}
 
+	_createIdentifier() {
+		const orderedConcatenationListIdentifiers = this._orderedConcatenationList.map(info => {
+			switch(info.type) {
+				case "concatenated":
+					return info.module.identifier();
+			}
+		}).filter(Boolean).join(" ");
```

Comment:
Instead of mapping, filtering, and joining, I wonder if this would be faster and use less memory if you call hash.update while iterating in a forEach:

```js
const hash = crypto.createHash("md5");
this._orderedConcatenationList.forEach(info => {
  if (info.type === "concatenated") {
    hash.update(info.module.identifier() + " ");
  }
});
return this.rootModule.identifier() + " " + hash.digest("hex");
```

Reply:
Isn't `sha-1` generally faster than `md5` these days?

Reply:
Not sure, but the difference is probably negligible. Profiling should tell us either way.

If we want raw speed, md4 might be worth considering as well. Not sure about collisions though.

Reply:
Oh, I've also had good success using string-hash in client code. https://github.com/darkskyapp/string-hash

Reply:
Splitting hairs here but we don't really need a crypto hash. 

We need to not have collisions & given it's purpose is purely performance, it needs to be fast. 

Medium to the rescue: https://medium.com/@drainingsun/in-search-of-a-good-node-js-hashing-algorithm-8052b6923a3b

It might be worth at least considering the use of a non-crypto hashing library given in at least some cases, the performance gains are significant.

Reply:
We are going to do some testing around hashing algos & if we go with a non-crypto hash, it will be changed everywhere.

There are other places in webpack where MD5 is leveraged that could benefit from a non-crypto algorithm.

Reply:
That sounds good to me. I think in that case, this PR should stick with md5 and then can be upgraded with whatever you land on after that testing is complete.

I'd love to try this out at Airbnb ASAP--is there any chance we can get this in and get a new webpack 3 release out?

Reply:
My only reasoning for picking this algorithm was consistency really. It is what NormalModule is using. Happy to change it to whatever.

Reply:
Cleaned out the comments that really have no bearing on this particular PR.

@lencioni - I do believe that's the plan though webpack/webpack isn't my domain. I'm going to work on the hash testing now. Assuming the results are positive, i'll tag you in that PR as well given it's probably going to be applicable to airbnb's perf issues as well.

Reply:
@filipesilva - Leave it the way it is, we kind of went off on a tangent that is likely to lead to more perf improvements but needs to be done across the entire project.

Reply:
Cleanup: https://github.com/webpack/webpack/pull/6001